homeassistant/components/template/light.py

```diff
@@ -335,3 +391,81 @@ def template_light_startup(event):
                         self._name,
                         ex,
                     )
+
+    async def update_temperature(self):
+        """Update the temperature from the template."""
+        if self._temperature_template is not None:
+            try:
+                temperature = int(self._temperature_template.async_render())
+            except TemplateError as ex:
+                _LOGGER.error(ex)
+                self._temperature = None
+
+            if 0 <= temperature <= self.max_mireds:
+                self._temperature = temperature
+            else:
+                _LOGGER.error(
+                    "Received invalid color temperature : %s. Expected: 0-%s",
+                    temperature,
+                    self.max_mireds,
+                )
+                self._temperature = None
+
+    async def update_brightness(self):
+        """Update the brightness from the template."""
+        if self._level_template is not None:
+            try:
+                brightness = self._level_template.async_render()
+            except TemplateError as ex:
+                _LOGGER.error(ex)
+                self._state = None
+
+            if 0 <= int(brightness) <= 255:
+                self._brightness = int(brightness)
+            else:
+                _LOGGER.error(
+                    "Received invalid brightness : %s. Expected: 0-255", brightness
+                )
+                self._brightness = None
+
+    async def update_color(self):
+        """Update the hs_color from the template."""
+        if self._color_template is not None:
+            hs_color = None
+            self._color = None
+
+            try:
+                hs_color = make_tuple(self._color_template.async_render())
```

Comment:
We should not do it like this. Instead parse it, `h_str, s_str = value.split(',', 1)`

Reply:
Fixed.